dts/arm/nordic/nrf52840.dtsi

```diff
@@ -13,8 +13,18 @@
 		};
 	};
 
-	flash0: flash@0 {
-		reg = <0x00000000 DT_FLASH_SIZE>;
+	flash-controller@0x4001E000 {
+			compatible = "nrf,nrf52-flash-controller";
+			reg = <0x4001E000 0x550>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash {
+				compatible = "flash";
+				reg = <0x00000000 DT_FLASH_SIZE>;
+				write-block-size = <4>;
```

Comment:
Curious where you expect to end up using write-block-size in actual code.  I wonder if such properties belong up with the flash-controller node instead.